agenda/presentation/src/main/java/com/gkp/agenda/presentation/edit/EditAgendaItemViewModel.kt

```diff
@@ -46,41 +40,29 @@ class EditAgendaItemViewModel(
 
     init {
         uiState = uiState.copy(agendaItemType = agendaItemType)
-        snapshotFlow {
-            agendaItemId
-        }
-            .filterNotNull()
-            .flatMapLatest {
-                when (agendaItemType) {
-                    AgendaItemType.REMINDER -> agendaRepository.fetchAgendaItemReminder(it)
-                    AgendaItemType.TASK -> agendaRepository.fetchAgendaItemTask(it)
-                    AgendaItemType.EVENT -> agendaRepository.fetchAgendaItemEvent(it)
-                }
-            }
-            .onEach { result ->
-                val agendaItemTask = result.getDataOrNull()
-                agendaItemTask?.let {
-                    uiState = uiState.copy(
-                        id = it.id,
-                        title = it.title,
-                        description = it.description ?: "",
-                        dateTime = it.time.toLocalDateTime(),
-                        reminderTextId = reminderLongToReminderTimeTextId(
-                            it.remindAt,
-                            it.time
-                        ),
-                        editTitleTextState = TextFieldState(
-                            it.title
-                        ),
-                        editDescriptionTextState = TextFieldState(
-                            it.description ?: ""
-                        )
+        agendaItemId?.let {
+            viewModelScope.launch {
+                val agendaItem = agendaRepository.getAgendaItemForId(it)
+                uiState = uiState.copy(
```

Comment:
Something to think about when working on real-world projects is the following:

If, for some reason, the `agendaItemId` is null, then your app’s screen state will not update. The user will not be able to tell that something is wrong. The ID of the item is a crucial part of this screen and should not be null, and if it is, then there is a bug that needs to be fixed.